.github/workflows/ci-build.yaml

```
@@ -401,7 +401,15 @@ jobs:
       - run: yarn --cwd ui test
       - run: yarn --cwd ui lint
       - run: yarn --cwd ui deduplicate
-      - run: git diff --exit-code
+      - if: github.actor == 'dependabot[bot]'
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+          git add ui/yarn.lock
+          git commit -m 'chore: update yarn.lock'
```

```suggestion
          git commit -s -m 'chore: deduplicate yarn.lock'
```

be more explicit in the message and add a sign-off

---

this line might [also need](https://github.com/foxglove/studio/blob/0977e3f553d0fc7bed19eabfad477536c32c6b96/.github/workflows/dependabot-fix.yml#L58) a `--allow-empty` or a conditional to otherwise check for empty

actually a conditional might(?) be necessary, otherwise this could cause an infinite loop as written (new commit -> rebuild -> repeat). maybe not if the `github.actor` check recognizes that commit is separate?

---

If GITHUB_TOKEN, which is provided by default in actions, is used, github actions are not triggered.
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow

---

> this line might [also need](https://github.com/foxglove/studio/blob/0977e3f553d0fc7bed19eabfad477536c32c6b96/.github/workflows/dependabot-fix.yml#L58) a `--allow-empty` or a conditional to otherwise check for empty

Yea the lack of conditional and `--allow-empty` is causing failures in CI when there are no changes, see https://github.com/argoproj/argo-workflows/pull/12880#issuecomment-2036064662

---

> @Yaminyam if you'd like to work on a more optimal fix that checks if there are any `yarn.lock` before running any of this code, that'd be great!

So I added this logic in https://github.com/agilgur5/argo-workflows/pull/3, so this has been completed.
Also due to the [below comment](https://github.com/argoproj/argo-workflows/pull/12234#discussion_r1552636495), this entire approach is not workable, so I'm going to be reverting this shortly.